public/app/plugins/datasource/cloudwatch/language_provider.ts

```diff
@@ -126,124 +86,70 @@ export class CloudWatchLanguageProvider extends LanguageProvider {
       (token: any) =>
         token.offsets.start <= value!.selection?.start?.offset && token.offsets.end >= value!.selection?.start?.offset
     )[0];
-    const isFirstToken = curToken.prev === null || curToken.prev === undefined;
-    const prevToken = prevNonWhitespaceToken(curToken);
-    const funcsWithFieldArgs = [
-      'avg',
-      'count',
-      'count_distinct',
-      'earliest',
-      'latest',
-      'sortsFirst',
-      'sortsLast',
-      'max',
-      'min',
-      'pct',
-      'stddev',
-      'ispresent',
-      'fromMillis',
-      'toMillis',
-      'isempty',
-      'isblank',
-      'isValidIp',
-      'isValidIpV4',
-      'isValidIpV6',
-      'isIpInSubnet',
-      'isIpv4InSubnet',
-      'isIpv6InSubnet',
-    ].map(funcName => funcName.toLowerCase());
-
-    if (curToken.content === '(' && prevToken != null) {
-      if (funcsWithFieldArgs.includes(prevToken.content.toLowerCase()) && prevToken.types.includes('function')) {
-        const suggs = await this.getFieldCompletionItems(context?.logGroupNames ?? []);
-        return suggs;
-      }
-    }
 
-    // if (prevToken === null) {
-    //   return {
-    //     suggestions: [],
-    //   };
-    // }
-
-    // if (prevToken) {
-    //   console.log(`Previous token: '${prevToken.content}'`);
-    // }
+    const isFirstToken = !curToken.prev;
+    const prevToken = prevNonWhitespaceToken(curToken);
 
     const isCommandStart = isFirstToken || (!isFirstToken && prevToken?.types.includes('command-separator'));
-    //console.log(`Is command start? ${isCommandStart}`);
     if (isCommandStart) {
       return this.getCommandCompletionItems();
-    } else if (!isFirstToken) {
-      if (prevToken?.types.includes('keyword')) {
-        return this.handleKeyword(prevToken, context);
-      }
+    }
```

Comment:
The conditionals are flattened here so you can see case by case what will be suggested